packmods_osx.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# This script will pack all mods and write them to an output directory.
+# On first execution, the script will ask for the Starbound install location and optionally store it for future use.
+
+function help {
+        echo "help            - Display this message and exit."
+        echo "nosavedlocation - Prompt for a location instead of using the saved location."
+        exit
+}
+
+function get_asset_packer_location {
+        # If a location is stored and the user didn't specify not to use it, use it.
+        if [ -f .starboundlocation.txt ] && [ "${1,,}" != "nosavedlocation" ]; then
+                sb_loc=$(cat .starboundlocation.txt)
+        else
+                # Otherwise prompt for a new location and store it.
+                read -p "Please enter the path to your Starbound installation: " sb_loc
+                echo ${sb_loc} > .starboundlocation.txt
+        fi
+}
+
+function pack_mods {
+        # Remove the output directory if it exists then create it.
+        if [ -d "output" ]; then
+                rm -r output
+        fi
+        mkdir output
+
+        for dir in ./mods/*; do
+                # Ensure we're only iterating over directories.
+                [ -d "${dir}" ] || continue
+                dirname=$(basename "${dir}")
+                "${sb_loc}/osx/asset_packer" "./${dir}" "./output/${dirname}.pak"
```

Comment:
This seems to be the only difference between this version for linux.
I'm not against the idea of having two different files, unless there is some important reason why we must have the two be separate.

For instance you could do something like:
```bash
for dir in ./mods/*; do
        # Ensure we're only iterating over directories.
        [ -d "${dir}" ] || continue
        dirname=$(basename "${dir}")
        if [[ "$OSTYPE"=="darwin"* ]]; then
                # OSTYPE corresponds to the linux operating system running
                # the script. For macs OSTYPE is "darwin"
                "${sb_loc}/osx/asset_packer" "./${dir}" "./output/${dirname}.pak"
        else
                # Otherwise it's linux
                "${sb_loc}/linux/asset_packer" "./${dir}" "./output/${dirname}.pak"
        fi
done
```
Then you wouldn't need to have two separate files.
Additionally renaming the single file to `packmods.sh` would be good.

Reply:
also unless my IDE is wrong, that syntax is valid and the information in the code-comments is correct that OSX is `"$OSTYPE"` is `"darwin"*` so you can just stitch that code into one of the two, rename the file, and delete the other one. ¯\_(ツ)_/¯